src/huggingface_hub/hub_mixin.py

```diff
@@ -0,0 +1,182 @@
+                generated when running :obj:`transformers-cli login` (stored in :obj:`~/.huggingface`).
+            revision(:obj:`str`, `optional`, defaults to :obj:`"main"`):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so ``revision`` can be any
+                identifier allowed by git.
+        .. note::
+            Passing :obj:`use_auth_token=True` is required when you want to use a private model.
+        """
+
+        model_id = pretrained_model_name_or_path
+        strict = kwargs.pop("strict", True)
+        map_location = kwargs.pop("map_location", torch.device("cpu"))
+        revision = kwargs.pop("revision", None)
+        force_download = kwargs.pop("force_download", False)
+        resume_download = kwargs.pop("resume_download", False)
+        proxies = kwargs.pop("proxies", None)
+        use_auth_token = kwargs.pop("use_auth_token", None)
+        cache_dir = kwargs.pop("cache_dir", None)
+        local_files_only = kwargs.pop("local_files_only", False)
+
+        if len(model_id.split("/")) == 1:
+            name = model_id
+        else:
+            _, name = model_id.split("/")
+
+        if name in os.listdir() and CONFIG_NAME in os.listdir(name):
+            print("LOADING weights from local directory")
+            config_file = os.path.join(name, CONFIG_NAME)
+        else:
+            try:
+                config_url = hf_hub_url(model_id, filename=CONFIG_NAME, revision=revision)
+                config_file = cached_download(config_url, cache_dir=cache_dir, force_download=force_download, proxies=proxies, resume_download=resume_download, local_files_only=local_files_only, use_auth_token=use_auth_token)
+            except:
+                config_file = None
+
+        if name in os.listdir():
+            print("LOADING weights from local directory")
+            model_file = os.path.join(name, PYTORCH_WEIGHTS_NAME)
+        else:
+            model_url = hf_hub_url(model_id, filename=PYTORCH_WEIGHTS_NAME, revision=revision)
+            model_file = cached_download(model_url, cache_dir=cache_dir, force_download=force_download, proxies=proxies, resume_download=resume_download, local_files_only=local_files_only, use_auth_token=use_auth_token)
+
+        if config_file is not None:
+            with open(config_file, "r", encoding="utf-8") as f:
+                config = json.load(f)
+            model = cls(config, **kwargs)
+        else:
+            model = cls(**kwargs)
+
+        state_dict = torch.load(model_file, map_location=map_location)
+        model.load_state_dict(state_dict, strict=strict)
+        model.eval()
+
+        return model
+
+    @staticmethod
+    def upload_to_hub(weights_directory: str, model_id: str, **kwargs):
```

Comment:
```suggestion
    def push_to_hub(weights_directory: str, model_id: str, **kwargs):
```

to be consistent with `Repository`'s method?